aws_signing_helper/file_system_signer.go

```diff
@@ -96,5 +116,126 @@ func GetFileSystemSigner(privateKey crypto.PrivateKey, certificate *x509.Certifi
 		return nil, "", errors.New("unsupported algorithm")
 	}
 
-	return FileSystemSigner{privateKey, certificate, certificateChain}, signingAlgorithm, nil
+	fsSigner := &FileSystemSigner{PrivateKey: privateKey, bundlePath: bundlePath, cert: certificate, certChain: certificateChain, certPath: certPath, pkcs12: pkcs12, privateKeyPath: privateKeyPath}
+	fsSigner.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		return nil, "", err
+	}
+	if certPath != "" {
+		fsSigner.watcher.Add(certPath)
+	}
+	if privateKeyPath != "" {
+		fsSigner.watcher.Add(privateKeyPath)
+	}
+	if bundlePath != "" {
+		fsSigner.watcher.Add(bundlePath)
+	}
+
+	if Debug {
+		log.Println("Starting file watcher")
+	}
+	go fsSigner.watch()
+
+	return fsSigner, signingAlgorithm, nil
+}
+
+func (fileSystemSigner *FileSystemSigner) watch() {
+	for {
+		select {
+		case event, ok := <-fileSystemSigner.watcher.Events:
+			// Channel is closed.
+			if !ok {
+				return
+			}
+
+			fileSystemSigner.handleEvent(event)
+
+		case err, ok := <-fileSystemSigner.watcher.Errors:
+			// Channel is closed.
+			if !ok {
+				return
+			}
+
+			log.Printf("Certificate watch error: %s", err)
+		}
+	}
+}
+
+func (fileSystemSigner *FileSystemSigner) handleEvent(event fsnotify.Event) {
+	if !(isWrite(event) || isRemove(event) || isCreate(event)) {
+		return
+	}
+
+	if Debug {
+		log.Printf("Certificate event :%v", event)
+	}
+
+	if isRemove(event) {
+		if err := fileSystemSigner.watcher.Add(event.Name); err != nil {
+			log.Printf("Error re-watching file: %s", err)
+		}
+	}
+
+	if event.Name == fileSystemSigner.certPath {
+		if fileSystemSigner.pkcs12 {
+			chain, privateKey, err := ReadPKCS12Data(fileSystemSigner.certPath)
+			if err != nil {
+				log.Fatalf("Failed to read modified PKCS12 certificate: %s", err)
```

Comment:
Minor, but I would prefer to write this in a way that is consistent with the rest of the codebase. The rest of the codebase does something like this: 
```
log.Println(msg)
os.Exit(1)
```

Reply:
I changed them.  FYI, there is a place that is using log.Fatal [here](https://github.com/aws/rolesanywhere-credential-helper/blob/main/aws_signing_helper/update.go#L39).